src/application/polymarket/utils.ts

```diff
@@ -35,3 +61,275 @@ export const isEventUrl = (url: string) => {
   const urlMatch = url.match(EVENT_URL_REGEX);
   return Boolean(urlMatch?.[0]);
 };
+
+export const buildL1SignaturePayload = (properties: {
+  address: Hex;
+  timestamp: string;
+}) => {
+  const { address, timestamp } = properties;
+  const domain = {
+    name: 'ClobAuthDomain',
+    version: '1',
+    chainId: BigInt(CHAIN.POLYGON.id),
+  };
+
+  const types = {
+    ClobAuth: [
+      { name: 'address', type: 'address' },
+      { name: 'timestamp', type: 'string' },
+      { name: 'nonce', type: 'uint256' },
+      { name: 'message', type: 'string' },
+    ],
+    EIP712Domain: [
+      {
+        name: 'name',
+        type: 'string',
+      },
+      {
+        name: 'version',
+        type: 'string',
+      },
+      {
+        name: 'chainId',
+        type: 'uint256',
+      },
+    ],
+  } as const;
+
+  const message = {
+    address: address,
+    timestamp,
+    nonce: BigInt(0),
+    message: 'This message attests that I control the given wallet',
```

Comment:
Just to make sure, was that message supposed to be a value or a comment?

Reply:
it was supposed to be a value, it is the one visible in metamask when you are asked to sign